### ISSUE TITLE
Feature request: Start/Stop widget/icon/button

### ISSUE THREAD

**Issue by claude-cp** [syncthing/syncthing-android#1578]
Most of the times I do not want the app to run continously.. all day, making 30-40 requests per minute.
In fact the opposite is true - when I intend to run it.

**It would be great if a simple widget or icon or button can be added to toggle syncthing on / off.** 

Such an example can be seen in [Drowser](https://gitlab.com/juanitobananas/drowser/tree/HEAD) which is an open source android app so the code can be checked there. (A quick search within the code there for "widget" shows some results that may be relevant?)

I am not sure if it actually is a widget class or what it is, as I am not an android developer, never have developed for Android, but with Drowser installed, you can tap-hold its icon (so as you would a widget, Android 10) and out of that create another icon like an app, which acts like a button for the app's main functionality, and thus now having a button you do not need to open the app itself anymore (90% of the times, as **can be** the case with Syncthing too).

I suspect that approach might turn out not too difficult to replicate here (?), hence bringing it up - because it looks to me like they used the most basic and straightforward way to enable such a feature.
Although, in this case for Syncthing, it would be amazing if the resulting icon/widget/button could change its visuals - so as to display the app's current state (on or off).

### Version Information
    App Version: 1.11.1
    Syncthing Version: v1.11.1
    Android Version: Android 10

**Comment by claude-cp** [comment]
Silly me, I just found out that Android simply calls that a [shortcut](https://images.techhive.com/images/article/2016/11/android-71-app-shortcuts-100693489-large.jpg) which you can drag onto the screen as if it were an app "icon". :)

**Comment by theraser** [comment]
That exactly what I wanted to suggest. This way you could automate some starting/stopping using Tasker, which can access the common app shortcuts.

I would love to see this feature, too. Completely killing and starting the app itself using Tasker works somehow, but will bring Syncthing to the foreground when starting.

I got the problem that Syncthing stops itself (?) after some time when inactive (i.e. when "only run while charging" is active and not plugged in) and won't activate again when plugged in again. Using Tasker I could work around this problem, but I need a shortcut to trigger a start in the background :+1:

**Comment by AudriusButkevicius** [comment]
You can already automate using tasker by sending an intent.

**Comment by claude-cp** [comment]
**However, Tasker for one, is not open source. I'd rather keep my ecosystem open source and this still means that Syncthing having its own shortcut is the way in this regard. **

Before suggesting perhaps an open source alternative to Tasker, I also want to point out: I don't think installing another app for this simple thing is the way, in general. 
A clean and minimal ecosystem is much preferred, rather than a bloated one, with apps depending on apps depending on apps. I know it's 2020, but I hope at least the open source community does not lose its way into the software ecosystem. 

Simplicity, robustness and self-reliance are virtues in themselves, and something I value highly in software especially.

**Comment by beerse** [comment]
For the same reason (and more) I have set syncthing to only be active while charging, and while at my home-wifi.
The only problem in this setting is that once I enable syncthing-gui while not charging, it asks me to alter the settings to allow it to run. Here I would prefer the option to overrule the setting for now until I close the interface.

**Comment by AudriusButkevicius** [comment]
Yes, but back in the day, there was no pause feature, so pausing was done by starting and stoppiing the application.

**Comment by beerse** [comment]
> Correct me if I'm wrong, but wouldn't a "Start/Stop Syncthing" feature rather have to pause/unpause all folders instead of turning off Syncthing itself ? Even in the Web UI you can't "Stop Syncthing" but there is a "Pause All" button.


I'm not sure but I do think/hope 'pause'  is stop/start and reboot safe: If you put a dedicated folder-sync in pause and stop/start syncthing or reboot your machine, the result after syncthing is started once again, the directory is still in pause.
